### PR TITLE
Adding new indirect selection pages

### LINKED ISSUE #4773: Example in `profiles.yml` for `indirect_selection`
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Two pages:
- https://docs.getdbt.com/docs/core/connect-data-platform/profiles.yml
- new page named something like https://docs.getdbt.com/reference/global-configs/indirect-selection or https://docs.getdbt.com/reference/global-configs/selection, etc.

The new page should work in concert with these pre-existing page sections:
- https://docs.getdbt.com/reference/node-selection/test-selection-examples#indirect-selection
- https://docs.getdbt.com/reference/node-selection/test-selection-examples#syntax-examples
- https://docs.getdbt.com/reference/node-selection/yaml-selectors#indirect-selection

### What part(s) of the page would you like to see updated?

Starting in v1.8, https://github.com/dbt-labs/dbt-core/pull/9040 adds `indirect_selection` as a global config, so we should:

1. create a global config page for this flag
2. update [`profiles.yml`](https://docs.getdbt.com/docs/core/connect-data-platform/profiles.yml) accordingly

If we were to _just_ do number 2) without doing 1), then the update might look like this (with applicable version blocks, of course):
```
  [indirect_selection](/reference/node-selection/test-selection-examples#indirect-selection): <eager | cautious | buildable | empty>
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by runleonarun**
closes #4773 

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/static/_redirects`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/reference/global-configs/indirect-selection.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+sidebar: "Indirect selection"
+---
+
+Supply the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests will be run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
+
+When all flags are set, CLI configurations take precedence, then environment variables, then user configurations. Read the docs on [global configs](/reference/global-configs/about-global-configs) for details.
+
+Options are `empty`, `buildable`, `cautious`, and `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Read more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
+
+For example, you can run tests that only refer to selected nodes using a CLI configuration:
```
Comment (runleonarun) [inline website/docs/reference/global-configs/indirect-selection.md:17]:
```suggestion
You can run tests that only refer to selected nodes using a CLI configuration, for example:
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+```text
+
+$ export DBT_INDIRECT_SELECTION=cautious
+dbt run
+
+```
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using `profiles.yml` user configurations:
```
Comment (runleonarun) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
You can run tests that only refer to selected nodes using `profiles.yml` user configurations, for example:
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+<File name='Usage'>
+
+```text
+dbt test --indirect-selection cautious
+```
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using an environment variable example:
```
Comment (runleonarun) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
You can run tests that only refer to selected nodes using an environment variable, for example:
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+$ export DBT_INDIRECT_SELECTION=cautious
+dbt run
+
+```
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using `profiles.yml` user configurations:
+
+dbt Core users can configure
```
Comment (runleonarun) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
```

File: website/dbt-versions.js [hunk 5]
```diff
@@ -30,6 +30,10 @@ exports.versions = [
 ]
 
 exports.versionedPages = [
+  {
+    "page": "/reference/global-configs/indirect-selection",
```
Comment (runleonarun) [inline website/dbt-versions.js]:
```suggestion
    "page": "reference/global-configs/indirect-selection",
```

File: website/docs/docs/core/connect-data-platform/profiles.yml.md [hunk 6]
```diff
@@ -56,4 +57,4 @@ This section identifies the parts of your `profiles.yml` that aren't specific to
 
 ## User config
 
-You can set default values of global configs for all projects that you run using your local machine. See the docs on [global configs](/reference/global-configs/about-global-configs) for details.
+You can set default values of global configs for all projects that you run using your local machine. Read the docs on [global configs](/reference/global-configs/about-global-configs) for details.
```
Comment (nghi-ly) [inline website/docs/docs/core/connect-data-platform/profiles.yml.md]: sugg: more inclusive word choice for less abled people

```suggestion
You can set default values of global configs for all projects that you run using your local machine. Refer to [About global configs](/reference/global-configs/about-global-configs) for details.
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
+---
+title: "Indirect selection"
+id: "indirect-selection"
+sidebar: "Indirect selection"
+---
+
+Supply the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests will be run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
```
Comment (nghi-ly) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+When all flags are set, CLI configurations take precedence, then environment variables, then user configurations. Read the docs on [global configs](/reference/global-configs/about-global-configs) for details.
```
Comment (nghi-ly) [inline website/docs/reference/global-configs/indirect-selection.md]: sugg: for improved scannability/readability 

```suggestion
When all flags are set, the order of precedence are as follows. Refer to [About global configs](/reference/global-configs/about-global-configs) for details.

1. CLI configurations
1. Environment variables
1. User configurations
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+Options are `empty`, `buildable`, `cautious`, and `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Read more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
```
Comment (nghi-ly) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
You can set the flag to: `empty`, `buildable`, `cautious`, or `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Learn more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+You can run tests that only refer to selected nodes using a CLI configuration, for example:
```
Comment (nghi-ly) [inline website/docs/reference/global-configs/indirect-selection.md]: sugg: move "for example" to the front

File: website/docs/reference/global-configs/indirect-selection.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+<File name='Usage'>
+
+```text
+dbt test --indirect-selection cautious
+```
+
+</File>
+
+You can run tests that only refer to selected nodes using an environment variable, for example:
```
Comment (nghi-ly) [inline website/docs/reference/global-configs/indirect-selection.md]: same sugg

File: website/docs/reference/global-configs/indirect-selection.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+```text
+
+$ export DBT_INDIRECT_SELECTION=cautious
+dbt run
+
+```
+
+</File>
+
+You can run tests that only refer to selected nodes using `profiles.yml` user configurations, for example:
```
Comment (nghi-ly) [inline website/docs/reference/global-configs/indirect-selection.md]: same sugg

File: website/docs/reference/global-configs/indirect-selection.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+When all flags are set, CLI configurations take precedence, then environment variables, then user configurations. Read the docs on [global configs](/reference/global-configs/about-global-configs) for details.
+
+Options are `empty`, `buildable`, `cautious`, and `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Read more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
+
+You can run tests that only refer to selected nodes using a CLI configuration, for example:
+
+<File name='Usage'>
+
+```text
```
Comment (nghi-ly) [inline website/docs/reference/global-configs/indirect-selection.md]: is this example in a file? should it be `shell` instead of `text` ? i might be misunderstanding tho

**Review by nghi-ly — COMMENTED** [review]
looks great! left some comments, none are blocking

**Review by nghi-ly — APPROVED** [review]

File: website/docs/reference/global-configs/indirect-selection.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+When all flags are set, the order of precedence are as follows. Refer to [About global configs](/reference/global-configs/about-global-configs) for details.
+
+1. CLI configurations
+1. Environment variables
+1. User configurations
+
+You can set the flag to: `empty`, `buildable`, `cautious`, or `eager` (default). By default, dbt indirectly selects all tests if they touch any resource you select. Learn more about these options in [Indirect selection in Test selection examples](/reference/node-selection/test-selection-examples?indirect-selection-mode=empty#indirect-selection).
+
+You can run tests that only refer to selected nodes using a CLI configuration, for example:
```
Comment (runleonarun) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
For example, you can run tests that only refer to selected nodes using a CLI configuration:
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+<File name='Usage'>
+
+```text
```
Comment (runleonarun) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
```shell
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+dbt test --indirect-selection cautious
+```
+
+</File>
+
+You can run tests that only refer to selected nodes using an environment variable, for example:
```
Comment (runleonarun) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
For example, you can run tests that only refer to selected nodes using an environment variable
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+```text
+
+$ export DBT_INDIRECT_SELECTION=cautious
+dbt run
+
+```
+
+</File>
+
+You can run tests that only refer to selected nodes using `profiles.yml` user configurations, for example:
```
Comment (runleonarun) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
For example, you can run tests that only refer to selected nodes using `profiles.yml` user configurations:
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
+---
+title: "Indirect selection"
+id: "indirect-selection"
+sidebar: "Indirect selection"
+---
+
+Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
+
+When all flags are set, the order of precedence are as follows. Refer to [About global configs](/reference/global-configs/about-global-configs) for details.
```
Comment (matthewshaver) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
When all flags are set, the order of precedence is as follows. Refer to [About global configs](/reference/global-configs/about-global-configs) for more details:
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+
+<File name='Usage'>
+
+```shell
+dbt test --indirect-selection cautious
+```
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using an environment variable
```
Comment (matthewshaver) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
Or you can run tests that only refer to selected nodes using an environment variable:
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+```text
+
+$ export DBT_INDIRECT_SELECTION=cautious
+dbt run
+
+```
+
+</File>
+
+For example, you can run tests that only refer to selected nodes using `profiles.yml` user configurations:
```
Comment (matthewshaver) [inline website/docs/reference/global-configs/indirect-selection.md]: Each of these sections starts with "For example" and it feels a little strange to read out loud. Maybe we can vary it a bit?
```suggestion
You can also run tests that only refer to selected nodes using `profiles.yml` user configurations:
```

File: website/docs/docs/core/connect-data-platform/profiles.yml.md
Comment (dbeatty10) [inline website/docs/docs/core/connect-data-platform/profiles.yml.md:1]: I _think_ is moving to `dbt_project.yml` within https://github.com/dbt-labs/docs.getdbt.com/pull/4814

So we'll either need to merge that one first or merge this one first.

Then deal with any merge conflicts that come up.

File: website/docs/reference/global-configs/indirect-selection.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
+---
+title: "Indirect selection"
+id: "indirect-selection"
+sidebar: "Indirect selection"
+---
+
+Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `config:` block of `profiles.yml`, which sets default values for all projects.
```
Comment (dbeatty10) [inline website/docs/reference/global-configs/indirect-selection.md]: I _think_ is moving to `dbt_project.yml` within https://github.com/dbt-labs/docs.getdbt.com/pull/4814

So we'll want to update to `dbt_project.yml` (after confirming that it actually works, of course).

File: website/docs/reference/global-configs/indirect-selection.md (hunk 20 above)
Comment (dbeatty10) [inline website/docs/reference/global-configs/indirect-selection.md]: I _think_ is moving to `dbt_project.yml` within https://github.com/dbt-labs/docs.getdbt.com/pull/4814

So we'll want to update to `dbt_project.yml` (after confirming that it actually works, of course).

**Review by dbeatty10 — COMMENTED** [review]
Overall, looks good to me.

Two things:
1. I _think_ this flag is moving to `dbt_project.yml` within [#4814](https://github.com/dbt-labs/docs.getdbt.com/pull/4814), so we'll want to update to `dbt_project.yml` where applicable.
2. We'll need to confirm that it actually works within `dbt_project.yml`, of course.

File: website/docs/docs/core/connect-data-platform/profiles.yml.md
Comment (runleonarun) [reply website/docs/docs/core/connect-data-platform/profiles.yml.md:1]: @dbeatty10 I think I might change the base branch to #4814 and deal with the merge conflicts now.

File: website/docs/docs/core/connect-data-platform/profiles.yml.md [hunk 22]
```diff
@@ -28,6 +28,7 @@ This section identifies the parts of your `profiles.yml` that aren't specific to
   [debug](/reference/global-configs/logs#log-level): <true | false>
   [version_check](/reference/global-configs/version-compatibility): <true | false>
   [fail_fast](/reference/global-configs/failing-fast): <true | false>
+  [indirect_selection](/reference/global-configs/indirect-selection): <eager | cautious | buildable | empty>
```
Comment (runleonarun) [inline website/docs/docs/core/connect-data-platform/profiles.yml.md:39]: We might want to move all of these flags into the project file

File: website/docs/reference/global-configs/indirect-selection.md (hunk 21 above)
Comment (dbeatty10) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify. You can set this as a CLI flag or an environment variable. In dbt Core, you can also configure user configurations in [YAML selectors](/reference/node-selection/yaml-selectors) or in the `flags:` block of `dbt_project.yml`, which sets project-level flags.
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+```text
+
+$ export DBT_INDIRECT_SELECTION=cautious
+dbt run
+
+```
+
+</File>
+
+You can also run tests that only refer to selected nodes using `profiles.yml` user configurations:
```
Comment (dbeatty10) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
You can also run tests that only refer to selected nodes using `dbt_project.yml` project-level flags:
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+
+<File name='profiles.yml'>
```
Comment (dbeatty10) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
<File name='dbt_project.yml'>
```

File: website/docs/reference/global-configs/indirect-selection.md [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+
+```yaml
+
+config:
```
Comment (dbeatty10) [inline website/docs/reference/global-configs/indirect-selection.md]:
```suggestion
flags:
```

**Review by dbeatty10 — APPROVED** [review]
Left a few suggestions in light of https://github.com/dbt-labs/docs.getdbt.com/pull/4814.

But we can also make these updates later too since user configs will still be a backwards compatible option within v1.8.

File: website/docs/docs/core/connect-data-platform/profiles.yml.md (hunk 22 above)
Comment (dbeatty10) [reply website/docs/docs/core/connect-data-platform/profiles.yml.md:39]: Good call!

It's covered in #4814 ([preview](https://docs-getdbt-com-git-jerco-flags-behavior-changes-dbt-labs.vercel.app/reference/dbt_project.yml)) with a link to `<global-configs>` like this:

image
